### ISSUE TITLE
Stash cannot be re-used in multiple downstream gitlab-ci jobs?

### ISSUE THREAD

**Issue by pmpfr** [romanowski/hoarder#55]
Fairly simple gitlab-ci setup:
1 writer job to populate the cache (`sbt stash`)
2 later reader jobs try to read from it (`sbt stashApply`)

Problem: one of the two later jobs sees the stash, the other doesn't.

Edited highlights from the logs:

Writer:
```
Not downloading cache 2f98e9f0bb85a5d8564f60bc58e8e36655648eaa due to policy
$ set -eu
$ sbt clean test:compile stash Common/test
...
[info] Project BaggingAPI stashed to
/builds/vector-experiments/scala-vector-tests/sbt-stash/scala-vector-tests/HEAD/2.11
using classes from List(
ExportedCache(/builds/vector-experiments/scala-vector-tests/sbt-stash/scala-vector-tests/HEAD/2.11/apis/BaggingAPI/compile/analysis.zip,
  Some(/builds/vector-experiments/scala-vector-tests/sbt-stash/scala-vector-tests/HEAD/2.11/apis/BaggingAPI/compile/classes.zip)),
ExportedCache(/builds/vector-experiments/scala-vector-tests/sbt-stash/scala-vector-tests/HEAD/2.11/apis/BaggingAPI/test/analysis.zip,
  Some(/builds/vector-experiments/scala-vector-tests/sbt-stash/scala-vector-tests/HEAD/2.11/apis/BaggingAPI/test/classes.zip))
  )
...
Creating cache 2f98e9f0bb85a5d8564f60bc58e8e36655648eaa...
sbt-stash/: found 210 matching files               
Created cache
```

One reader works:
```
Checking cache for 2f98e9f0bb85a5d8564f60bc58e8e36655648eaa...
Successfully extracted cache
$ set -eu
$ sbt stashApply 'SomeProject/testOnly blah blah'
...
[info] Stash for BaggingAPI applied from
/builds/vector-experiments/scala-vector-tests/sbt-stash/scala-vector-tests/HEAD/2.11
to List(
  /builds/vector-experiments/scala-vector-tests/apis/BaggingAPI/target/scala-2.11/classes,
  /builds/vector-experiments/scala-vector-tests/apis/BaggingAPI/target/scala-2.11/test-classes
)
```

One reader fails:
```
Checking cache for 2f98e9f0bb85a5d8564f60bc58e8e36655648eaa...
Successfully extracted cache
$ set -eu
$ sbt stashApply 'SomeOtherProject/testOnly blah blah'
...
[error] (BaggingAPI/*:stashApply) java.lang.AssertionError: assertion failed: Cache does not exists in
 /builds/vector-experiments/scala-vector-tests/sbt-stash/scala-vector-tests/HEAD/2.11/apis/BaggingAPI/compile
```

This is from a multi-module build.  I am overriding `globalStashLocation` to a place where gitlab-ci cache can use it: 
```
globalStashLocation := (baseDirectory in ThisBuild).value / "sbt-stash"
```

**Comment by romanowski** [comment]
Hello @pmpfr, is your gitlab project open so I can take a look?

If not can you confirmed that gitlab actually cached hoarder caches? When I debugged travis caches integration printed all cached dirs using `ls -alR <cache-dir>` on both export and import job

I will try to look at gitlab integration (#56) so I you can use more gitlab friendly setup.

**Comment by pmpfr** [comment]
I'm sorry, I was working on this again yesterday and I'm fairly convinced it's not an sbt-hoarder issue.  I was planning to update & close this issue when I've finally got it working.  I was confused by a combination of a build file issue (which caused sbt to always want to recompile some stuff in each new session for me which I've now fixed) and numerous gitlab-ci issues, all masking each other.

Latest from debugging with ls -R in gitlab-ci last night probably shows the directory is not getting uncached as I expect (on the "import job").
